### PR TITLE
Fix handling of service dates around DST, update Timex to 3.7

### PR DESCRIPTION
**Pull request by bklebe**
[fix handling of service dates around DST](https://app.asana.com/0/584764604969369/1201772965576880/f)

The algorithm is essentially copied from [RTR's version](https://github.com/mbta/rtr/blob/master/rtr/lib/rtr/utilities/time.ex#L120) except it uses 3am as the cutoff instead of 2am. Also bumped Timex to 3.7 since that's what uncovered the bug originally.

### CODE REVIEW

File: apps/train_loc/lib/train_loc/utilities/time.ex [hunk 1]
```diff
@@ -48,16 +48,18 @@ defmodule TrainLoc.Utilities.Time do
...
       |> in_local_tz
-      |> Timex.shift(hours: -3)
-      |> case do
-        %DateTime{} = dt -> dt
-        %Timex.AmbiguousDateTime{before: before} -> before
-      end
+      |> DateTime.to_date()
 
-    DateTime.to_date(dt)
+    local_time = in_local_tz(current_time)
```
Comment (paulswartz) [inline apps/train_loc/lib/train_loc/utilities/time.ex:56]: Suggestion: you're calling `in_local_tz` twice: can you get `current_date` from `local_time` instead?

File: apps/train_loc/test/train_loc/encoder/vehicle_positions_enhanced_test.exs [hunk 2]
```diff
@@ -151,7 +151,7 @@ defmodule TrainLoc.Encoder.VehiclePositionsEnhancedTest do
       assert start_date(dt) == "20180322"
     end
 
-    test "converts UTC datestimes into service data around DST transitions" do
+    test "converts UTC datestimes into service dates around DST transitions" do
```
[the remark this answers is not in the document]
Comment (bklebe) [reply apps/train_loc/test/train_loc/encoder/vehicle_positions_enhanced_test.exs:154]: Yes, this test fails with Timex 3.7 without the change.

[the remark this answers is not in the document]
Comment (bklebe) [reply apps/train_loc/test/train_loc/encoder/vehicle_positions_enhanced_test.exs:154]: Actually, that probably means that these commits should be squashed. Technically the tests would be broken on that commit

**Review by paulswartz — APPROVED** [review]
🍰

**Pull request merged**